ical.go

```diff
@@ -498,6 +498,47 @@ func (props Props) SetRecurrenceRule(rule *rrule.ROption) {
 	}
 }
 
+func (props Props) RecurrenceSet(loc *time.Location) (*rrule.Set, error) {
+	roption, err := props.RecurrenceRule()
+	if err != nil {
+		return nil, fmt.Errorf("ical: error parsing Recurrence: %v", err)
+	}
+	dateTime, err := props.DateTime(PropDateTimeStart, loc)
+	if err != nil {
+		return nil, fmt.Errorf("ical: error parsing dtstart: %v", err)
+	}
+
+	rule, err := rrule.NewRRule(*roption)
+	if err != nil {
+		return nil, fmt.Errorf("ical: error buildling RRULE: %v", err)
+	}
+
+	ruleSet := rrule.Set{}
+	ruleSet.RRule(rule)
+	ruleSet.DTStart(dateTime)
+
+	if exdates := props[PropExceptionDates]; exdates != nil {
```

Comment:
This `if` can be omitted:

```go
for _, exdateProp := range props[PropExceptionDates] {
```

Reply:
I'm rather confused as to why this works.

Reply:
Accessing a non-existing map key returns nil. A nil slice is a perfectly valid 0-sized slice.

Reply:
Makes sense, it's just that trying to run `for _, _ := range nil {}` doesn't work, which is somewhat unexpected.

Reply:
That'd be because Go can't infer the type of the slice elements. Try this:

```go
for _, _ = range []int(nil) {}
```

Reply:
Interesting, thanks!